### ISSUE TITLE
numpy memory issue 

### ISSUE THREAD

**Issue by remotejob** [WillKoehrsen/recurrent-neural-networks#2]
Excellent job! I try use it for my project but unfortunately my corpus quit big 400000 words and 4000000 sequences so  np.zeros((400000,4000000), dtype=np.int8) don't work --> MemoryError. Biggest PC I can use have 51GB memory anyway it not enough. My question: how possible resolve memory issue ? Partial training ? (split corpus on small peaces?). 
Thanks.

**Comment by WillKoehrsen** [comment]
You could split your corpus and train on fewer sequences at a time. However, your splits would have to be very small (a low number of sequences) if you still want to `one_hot_encode` the labels. 

Another option would be to _not_ `one_hot_encode` the targets. Instead, keep them as integers and then use `keras.losses.sparse_categorical_crossentropy(y_true, y_pred)` as the loss. 

I'm not sure if the performance would be significantly affected using integer targets compared to one-hot-encoding. In either case, you're still going to have to learn word embeddings for all 400,000 words (or download pre-trained embeddings) which could pose a challenge.
